modules/dnn/src/nms.cpp

```diff
@@ -28,6 +30,28 @@ void NMSBoxes(const std::vector<Rect>& bboxes, const std::vector<float>& scores,
     NMSFast_(bboxes, scores, score_threshold, nms_threshold, eta, top_k, indices, rectOverlap);
 }
 
+static inline float rotatedRectIOU(const RotatedRect& a, const RotatedRect& b)
+{
+    std::vector<Point2f> inter, hull;
+    int res = rotatedRectangleIntersection(a, b, inter);
```

Comment:
I wonder if we can have a function like rotatedRectangleIntersectionArea(), which will compute area of the intersection without storing the coordinates explicitly. Another question - since a rotated rectangle is a convex polygon, its intersection is a convex polygon as well. Why is convexHull() called then?

Reply:
@vpisarev, that's right but now `rotatedRectangleIntersection` returns just intersection points without contour orientation.
In example, 
```cpp
std::vector<Point2f> inter;
RotatedRect r1(Point2f(110.48, 272.86), Size2f(34.1097, 12.6564), 0);
RotatedRect r2(Point2f(109.976, 274.791), Size2f(36.023, 13.1669), 0);
rotatedRectangleIntersection(r1, r2, inter);
for (int i = 0; i < inter.size(); ++i)
    std::cout << inter[i].x << " " << inter[i].y << std::endl;
```
gives
```
93.4252 268.208
127.535 268.208
93.4252 279.188
127.535 279.188

2 ---- 3
|      |
0 ---- 1
```
So it just orient them. Otherwise `contourArea` returns `0`.

Reply:
@vpisarev, Done. `rotatedRectangleIntersection` returns an ordered set of points now.